quinn/examples/insecure_connection.rs

```diff
@@ -0,0 +1,106 @@
+//! Demonstrates how to make a QUIC connection that ignores the server certificate.
+//!
+//! Run:
+//! ```
+//! $ cargo run --example insecure_connection --features="dangerous_configuration"
```

Comment:
How does this work? We don't expose such a feature ourselves.

Reply:
This feature is exposed by `rustls`. Since TLS is first class citizen of QUIC, I figured it's useful to have this example as well.

Reply:
Right, but how does `--features` know you're talking about rustls? Quinn itself has no such feature name.

Reply:
I believe cargo defines features for a whole project so they get passed to all libraries and their dependencies.

Reply:
Are you sure? It doesn't seem to work here:
```
> cargo run --example insecure_connection --features="dangerous_configuration"
error: Package `quinn v0.2.0 (/home/ralith/src/quinn/quinn)` does not have these features: `dangerous_configuration`
```

Additionally, this breaks `cargo test` because that doesn't know about the [required feature](https://doc.rust-lang.org/cargo/reference/manifest.html#the-required-features-field-optional).